web/src/helpers/hooks/useIdeState.js

```diff
@@ -36,14 +36,19 @@ export const useIdeState = () => {
             ? [...state.consoleMessages, payload.message]
             : payload.message,
         }
+      case 'setIdeType':
+        return {
+          ...state,
+          ideType: payload.message,
+        }
     }
   }
 
-  function dispatchMiddleware(dispatch) {
+  function dispatchMiddleware(dispatch, state) {
```

Comment:
It seems obvious in retrospect, but I'm really glad you thought of this.

Reply:
I'm gonna have to remember this for future projects, it seems like a muscular little pattern.